internal/provider/virtualmachine/resource_virtualmachine_constructor.go

```diff
@@ -173,6 +175,9 @@ func (c *Constructor) Setup() util.Processors {
 					vmBuilder.WaitForLease(interfaceName)
 				}
 				vmBuilder.NetworkInterface(interfaceName, interfaceModel, interfaceMACAddress, interfaceType, networkName)
+				if bootOrder != 0 {
```

Comment:
A valid bootOrder is > 0. We should also check whether the value is negative.

https://github.com/kubevirt/kubevirt/blob/8958e5558e65d3be1626d1eee56a89a80eda4e68/staging/src/kubevirt.io/api/core/v1/schema.go#L634-L639